libs/designsystem/src/lib/components/modal/services/modal-outlet.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+import {
+  Router,
+  ActivatedRoute,
+  ChildrenOutletContexts,
+  NavigationEnd,
+  RouterEvent,
+} from '@angular/router';
+import { filter } from 'rxjs/operators';
+
+import { ModalWrapperComponent } from '../modal-wrapper/modal-wrapper.component';
+
+@Injectable({ providedIn: 'root' })
+export class ModalOutlet {
+  resolve$ = this.router.events.pipe(
+    filter((event: RouterEvent) => event instanceof NavigationEnd),
+    filter((event: NavigationEnd) => {
```

Comment:
Consider using type guard instead of casting:
```suggestion
    filter((event): event is NavigationEnd => event instanceof NavigationEnd),
    filter((event) => {
```